src/dmd/backend/evalu8.d

```diff
@@ -1876,4 +1898,51 @@ elem * evalu8(elem *e, goal_t goal)
   return e;
 }
 
-#endif /* !SPP */
+/******************************
+ * This is the same as the one in el.c, but uses native D reals
+ * instead of the soft long double ones.
+ */
+
+extern (D) targ_ldouble el_toldoubled(elem *e)
+{
+    targ_ldouble result;
+
+    elem_debug(e);
+    assert(e.Eoper == OPconst);
+    switch (tybasic(typemask(e)))
+    {
+        case TYfloat:
+        case TYifloat:
+            result = e.EV.Vfloat;
+            break;
+        case TYdouble:
+        case TYidouble:
+        case TYdouble_alias:
+            result = e.EV.Vdouble;
+            break;
+        case TYldouble:
+        case TYildouble:
+            result = e.EV.Vldouble;
+            break;
+        default:
+            result = 0;
+            break;
+    }
+    return result;
+}
+
+/***************************************
+ * Copy of _modulo from fp.c. Here to help with linking problems.
+ */
+version (CRuntime_Microsoft)
+{
+    extern (D) private real _modulo(real x, real y)
```

Comment:
I think the casts should go into this function instead of the call.

Reply:
all right